fix(remote): failure reason should be printed

bheesham opened the pull request:
Uses `fmt::print` to log failure messages. Results in a connection
failure being printed as:

        Unable to send request to 'http://localhost:9091/transmission/rpc/': Couldn't connect to server
Fixes #7010

tearfur reviewed (commented):
IMO this should use `fmt::print()` like all other lines. Having log levels and whatnot here adds no value.

bheesham commented:
@tearfur done! I changed the format of the connection error to be similar to the other messages as well (e.g. `error message: subject`).

mikedld commented on utils/remote.cc:
```diff
@@ -2397,7 +2397,7 @@ int flush(char const* rpcurl, tr_variant* benc, RemoteConfig& config)
     if (auto const res = curl_easy_perform(curl); res != CURLE_OK)
     {
-        tr_logAddWarn(fmt::format(" ({}) {}", rpcurl_http, curl_easy_strerror(res)));
+        fmt::print(stderr, "{}: {}\n", curl_easy_strerror(res), rpcurl_http);
```
Not sure what `curl_easy_strerror()` returns, but I think general approach is to prepend the failure with some description of our own, so maybe
```suggestion
        fmt::print(stderr, "Unable to send request to '{}': {}\n", rpcurl_http, curl_easy_strerror(res));
```

bheesham replied:
Ah, yeah, good call. Applied.

ckerr reviewed (commented):
Do the other calls to `tr_logAddWarn()` make sense in `utils/remote.cc`, or should they be removed too?

bheesham commented:
> Do the other calls to `tr_logAddWarn()` make sense in `utils/remote.cc`, or should they be removed too?

Solid call -- I rebased and replaced the other `tr_logAddWarn` with a print. I also went ahead and removed the include for the log header, since we're not using anything from there any more.

bheesham commented:
<details>
<summary>How this looks with <code>-b</code></summary>

```
; ./build/utils/transmission-remote localhost -b --list
posting:
--------
{"arguments":{"fields":["addedDate","error","errorString","eta","id","isFinished","leftUntilDone","name","peersGettingFromUs","peersSendingToUs","rateDownload","rateUpload","sizeWhenDone","status","uploadRatio"]},"method":"torrent-get","tag":6}
--------
* processing: http://localhost:9091/transmission/rpc/
* Couldn't find host localhost in the (nil) file; using defaults
*   Trying [::1]:9091...
* connect to ::1 port 9091 failed: Connection refused
*   Trying 127.0.0.1:9091...
* connect to 127.0.0.1 port 9091 failed: Connection refused
* Failed to connect to localhost port 9091 after 0 ms: Couldn't connect to server
* Closing connection
Unable to send request to 'http://localhost:9091/transmission/rpc/': Couldn't connect to server
```
</details>

<details>
<summary>How this looks <i>without</i> <code>-b</code></summary>

```
; ./build/utils/transmission-remote localhost --list
Unable to send request to 'http://localhost:9091/transmission/rpc/': Couldn't connect to server
```
</details>

<details>
<summary>How this looks <b>on main</b> with <code>-b</code></summary>

```
; ./build/utils/transmission-remote localhost -b --list
posting:
--------
{"arguments":{"fields":["addedDate","error","errorString","eta","id","isFinished","leftUntilDone","name","peersGettingFromUs","peersSendingToUs","rateDownload","rateUpload","sizeWhenDone","status","uploadRatio"]},"method":"torrent-get","tag":6}
--------
* processing: http://localhost:9091/transmission/rpc/
* Couldn't find host localhost in the (nil) file; using defaults
*   Trying [::1]:9091...
* connect to ::1 port 9091 failed: Connection refused
*   Trying 127.0.0.1:9091...
* connect to 127.0.0.1 port 9091 failed: Connection refused
* Failed to connect to localhost port 9091 after 0 ms: Couldn't connect to server
* Closing connection
```
</details>

<details>
<summary>How this looks <b>on main</b> <i>without</i> <code>-b</code></summary>

```
; ./build/utils/transmission-remote localhost --list
< empty >
```
</details>

tearfur commented:
@ckerr ping

ckerr commented:
Notes: Improved error logging.

bheesham commented:
Thanks @ckerr and @tearfur !
